src/node_buffer.cc

```diff
@@ -741,6 +749,45 @@ void SlowByteLengthUtf8(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(args[0].As<String>()->Utf8Length(env->isolate()));
 }
 
+#if NODE_HAS_SIMD_NEON
+uint32_t FastByteLengthUtf8(Local<Value> receiver,
+                            const v8::FastOneByteString& source) {
+  auto data = reinterpret_cast<const uint8_t*>(source.data);
+  auto length = source.length;
+
+  uint32_t result{0};
```

Comment:
Because result is a 32-bit integer and we are overcounting by a factor of 4, there might be overflow if a string exceeds a gigabyte or so. If you use a 64-bit counter, then you are more likely to be fine.

Reply:
V8 Fast API does not support returning uin64_t. Therefore, I don't think it will ever exceed the limit and overflow.

Reply:
@anonrig I meant using a 64-bit counter internally. I did not mean that you should change the function signature. Suppose you have a 2 GB input string made of the character `é`. Then you'd have an overflow.

(This is *nitpicking*. Your function is correct, and will only fail in really huge strings where you'd have other problems anyhow...)

Reply:
Maybe at least add an assertion?

Reply:
No, you can’t say “This won’t overflow because we can’t use a larger type”.

What you _can_ do is to use an `uint64_t` for `result`, then check before returning that it fits into an `uint32_t`. If you don’t want do to that, you could at least add something like `static_assert(String::kMaxLength < std::numeric_limits<uint32_t>::max() / 8);` (but I’m not sure if that assertion actually holds – the first approach here is definitely preferable).